hawk/src/main/java/com/orhanobut/hawk/Hawk.java

```diff
@@ -345,8 +348,29 @@ public Chain(int capacity) {
     public boolean commit() {
       return HAWK.storage.put(items);
     }
+  }
+
+  public static Map<String, ?> getAllSharedPrefs() {
+    if (HAWK.storage != null && HAWK.storage instanceof SharedPreferencesStorage) {
+      return (((SharedPreferencesStorage) HAWK.storage).getAll());
+    }
+    return null;
+  }
 
+  public static void registerHawkPrefsChangedListener(HawkPreferenceChangedListener listener) {
```

Comment:
same generic idea should apply here, we can add this event listener into the storage interface